### PR TITLE
[DO NOT MERGE] Tentative vfs-2.31.1 branch

### PR DESCRIPTION
**Pull request by derrickstolee**
For this maintenance release, I am just merging `v2.31.1.windows.1` into the `vfs-2.31.0` branch. There is only one minor conflict in `commit-graph.c` because we took the chunk-format API early.

### CODE REVIEW

**Review by dscho — APPROVED** [review]
That diff looks _awful_, due to all those `calloc()`-related changes. But I got through it, and your merge looks fine to me.

File: t/helper/test-gvfs-protocol.c [hunk 1]
```diff
@@ -1921,7 +1921,7 @@ static int service_loop(struct socketlist *socklist)
 	struct pollfd *pfd;
 	int i;
 
-	pfd = xcalloc(socklist->nr, sizeof(struct pollfd));
+	CALLOC_ARRAY(pfd, socklist->nr);
```
Comment (dscho) [inline t/helper/test-gvfs-protocol.c:1924]: 👍

**Comment by dscho** [comment]
Oh no. This looks like the `push` side of things has an incompletely-cached `git-sdk-64-minimal`? Darn. I am investigating.

**Comment by dscho** [comment]
[Yes](https://github.com/microsoft/git/runs/2218601656?check_suite_focus=true#step:3:13):

> ```
> Downloading git-sdk-64-minimal
> 250 items extracted
> 360 items extracted
> C:\Windows\System32\tar.exe --posix -z -cf cache.tgz -P -C D:/a/git/git --files-from manifest.txt
> Cache Size: ~38 MB (39976208 B)
> Cache saved successfully
> ```

A correct run would look like [this](https://github.com/microsoft/git/runs/2152181633?check_suite_focus=true#step:3:9):

> ```
> Downloading git-sdk-64-minimal
> 250 items extracted
> 500 items extracted
> 750 items extracted
> 1000 items extracted
> 1250 items extracted
> 1500 items extracted
> 1750 items extracted
> 2000 items extracted
> 2250 items extracted
> 2500 items extracted
> 2619 items extracted
> C:\windows\System32\tar.exe --posix -z -cf cache.tgz -P -C D:/a/git/git --files-from manifest.txt
> Cache Size: ~75 MB (78598492 B)
> Cache saved successfully
> ```

Hrm. I am not quite sure what is going wrong, as I thought I already handle all network failures in the Action, via retrying. But apparently there is a network failure that can slip through that we do not handle yet.

**Comment by dscho** [comment]
> I thought I already handle all network failures in the Action, via retrying. But apparently there is a network failure that can slip through that we do not handle yet.

I addressed this in https://github.com/git-for-windows/setup-git-for-windows-sdk/pull/54 (actually in https://github.com/git-for-windows/setup-git-for-windows-sdk/pull/53, but that PR targets v1 of the Action, while `microsoft/git` still uses v0).

I restarted the jobs and miraculously, they succeeded (but for a _different_ reason than the fixed Action: there is now a successful build of that tree, the PR build, therefore the CI build is skipped with success).

**Comment by derrickstolee** [comment]
I pushed `vfs-2.31.1` and a new tag. Thanks!

**Pull request closed without merging**